planner/core/logical_plan_builder.go

```diff
@@ -3328,7 +3329,9 @@ func (b *PlanBuilder) buildProjUponView(ctx context.Context, dbName model.CIStr,
 // every row from outerPlan and the whole innerPlan.
 func (b *PlanBuilder) buildApplyWithJoinType(outerPlan, innerPlan LogicalPlan, tp JoinType) LogicalPlan {
 	b.optFlag = b.optFlag | flagPredicatePushDown | flagBuildKeyInfo | flagDecorrelate
-	ap := LogicalApply{LogicalJoin: LogicalJoin{JoinType: tp}}.Init(b.ctx, b.getSelectOffset())
+	ap := LogicalApply{LogicalJoin: LogicalJoin{
+		JoinType:      tp,
+		cartesianJoin: true}}.Init(b.ctx, b.getSelectOffset())
```

Comment:
I prefer to do not regard `Apply` as `CARTESIAN JOIN`.

Reply:
I tried to remove it but failed in the cascade optimizer.
IMO, it's reasonable to set it as cartesian join, because of no equal conditions were set. Also, in the later process, this field will change, so it's acceptable.

Reply:
`explain select a = (select a from t2 where t1.b = t2.b order by a limit 1) from t1`, this cannot pass the test if cascades enabled, and we remove cartesianJoin at this step.